BGL/include/CGAL/boost/graph/Seam_mesh.h

```diff
@@ -954,19 +972,102 @@ class Seam_mesh
+  template<class InputIterator>
+  TM_halfedge_descriptor add_seams(InputIterator first, InputIterator last)
   {
-    TM_halfedge_descriptor tmhd;
+    // must have an even number of input vertices
+    assert(std::distance(first, last) % 2 == 0);
+
+    TM_halfedge_descriptor tmhd = boost::graph_traits<TM>::null_halfedge();
+    InputIterator it = first;
+
+    while(it!=last) {
+      TM_vertex_descriptor v1 = *(it++);
+      TM_vertex_descriptor v2 = *(it++);
+
+      if(!add_seam(v1, v2))
+        continue;
+
+      if(tmhd == boost::graph_traits<TM>::null_halfedge()) {
+        tmhd = CGAL::halfedge(CGAL::edge(v1, v2, tm).first, tm);
+      }
+    }
+
+    return tmhd;
+  }
+
+  /// Create new seams.
+  ///
+  /// A seam edge is described by a pair of integers. The integer index
+  /// of a vertex of the underlying mesh is given by its position
+  /// in the container `tm_vds`.
+  ///
+  /// \tparam VdContainer must be a model of <a href="http://en.cppreference.com/w/cpp/concept/SequenceContainer"><tt>SequenceContainer</tt></a> (that is, provide
+  ///         the functions: `operator[]` and `at()`).
+  ///
+  /// \returns one of the halfedges of the seam mesh that is on a seam.
+  ///
+  /// \pre The stream must contain an even number of values.
+  template<typename VdContainer>
+  TM_halfedge_descriptor add_seams(std::ifstream& in,
+                                   const VdContainer& tm_vds)
+  {
+    std::vector<TM_vertex_descriptor> seam_vertices;
+    std::size_t s, t;
+    while(in >> s >> t) {
+      seam_vertices.push_back(tm_vds.at(s));
+      seam_vertices.push_back(tm_vds.at(t));
+    }
+
+    return add_seams(seam_vertices.begin(), seam_vertices.end());
+  }
+
+  /// Create new seams.
+  ///
+  /// A seam edge is described by a pair of integers. The integer
+  /// index of a vertex of the underlying mesh is defined as its position when
+  /// iterating over the vertices of the underlying mesh with
+  /// `boost::graph<TM>::vertices()`.
+  ///
+  /// \returns one of the halfedges of the seam mesh that is on a seam.
+  ///
+  /// \pre The stream must contain an even number of values.
+  TM_halfedge_descriptor add_seams(std::ifstream& in)
+  {
+    std::vector<TM_vertex_descriptor> tm_vds;
+    build_TM_vertices_vector(tm_vds);
+    return add_seams(in, tm_vds);
+  }
+
+  /// Create new seams.
+  ///
+  /// A seam edge is described by a pair of integers. The integer index
+  /// of a vertex of the underlying mesh is given by its position
+  /// in the container `tm_vds`.
+  ///
+  /// \returns one of the halfedges of the seam mesh that is on a seam.
+  ///
+  /// \tparam VdContainer must be a model of <a href="http://en.cppreference.com/w/cpp/concept/SequenceContainer"><tt>SequenceContainer</tt></a> (that is, provide
+  ///         the functions: `operator[]` and `at()`).
+  ///
+  /// \pre filename should be the name of a CGAL selection file: edges are
+  ///      described by pairs of integers, on the third line of the file.
```

Comment:
I think you should not publicly document this one. CGAL file format are not clearly defined.